pkg/registry/generic/registry/store_test.go

```diff
@@ -546,6 +546,230 @@ func TestStoreDelete(t *testing.T) {
 	}
 }
 
+func TestStoreHandleFinalizers(t *testing.T) {
+	EnableGarbageCollector = true
```

Comment:
if you have to change a global flag during a test, then change code until you don't. :)